src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -285,10 +291,11 @@ internal sealed class SynthesizedComplexParameterSymbol : SynthesizedParameterSy
             int ordinal,
             RefKind refKind,
             DeclarationScope scope,
+            ConstantValue? defaultValue,
             string name,
             ImmutableArray<CustomModifier> refCustomModifiers,
             SourceComplexParameterSymbolBase? baseParameterForAttributes)
-            : base(container, type, ordinal, refKind, scope, name)
+            : base(container, type, ordinal, refKind, scope, defaultValue, name)
         {
             Debug.Assert(!refCustomModifiers.IsDefault);
             Debug.Assert(!refCustomModifiers.IsEmpty || baseParameterForAttributes is object);
```

Comment:
If this assertion failed, you will need to add `|| defaultValue is not null` (or `|| defaultValue is object`)

Reply:
Yes I initially switched over to `SourceComplexParameterSymbol` and forgot to update those assertions and a few other things, so I reverted. The assertion issue should be fixed now though